docs/api-reference/react/use-widget.md

```diff
@@ -0,0 +1,43 @@
+# useWidget
+
+The `useWidget` hook is used to create React wrappers for custom deck.gl widgets.
+
+```tsx
+import DeckGL, {useControl} from '@deck.gl/react';
+import {CompassWidget as VanillaCompassWidget, CompassWidgetProps} from '@deck.gl/widgets';
+
+export const CompassWidget = (props: CompassWidgetProps = {}) => {
+  const widget = useWidget(VanillaCompassWidget, props);
+  return null;
+};
+
+<DeckGL>
+  <CompassWidget/>
+</DeckGL>
+```
+
+See a full example [here](../../developer-guide/custom-widgets/react-widgets.md).
```

Comment:
`Vanilla` sounds so weird... And the fact that I can't easily think of a better name makes me feel this is an indication of the dual usage of the word "widget" here being more problematic than initially expected.

How about we reserve the word "widget" for actual widgets, and talk about wrapping widgets into react "components"? I realize it will required rewriting the docs a bit, but I think a ton of confusion can be avoided.
```suggestion
import {CompassWidget, CompassWidgetProps} from '@deck.gl/widgets';

export type CompassProps = CompassWidgetProps;

export const Compass = (props: CompassProps = {}) => {
  const widget = useWidget(CompassWidget, props);
  return null; 
};

<DeckGL>
  <CompassWidget/>
</DeckGL>
```

See a full example [here](../../developer-guide/custom-widgets/react-widgets.md).
```

Reply:
Yeah, that's a fair point. A Widget is just a hook. You could compose multiple widgets in one component.. widgets may or may not mount a DOM element. So yeah these components could really be anything.. we're just suggesting a usage.

I'll try reworking the PRs with this idea.

The open question I have is how to implement a widget nested in a JSX view.. 

```
<MapView id="minimap">
  <Compass />
</MapView>
```

I was originally going to inject a `viewId` prop, assuming all props were passed into the hook, or introduce something new like a ViewContext. I've been putting this off thinking the community react library might be a better solution.